### PR TITLE
[MBL-1458 pt 1] Add alert flag for PPO

### PR DESCRIPTION
**Pull request by stevestreza-ksr**
# 📲 What

This PR implements the alert flag for the card view for pledged projects, as well as the relevant subviews. It adds a view model to cover handling actions like tapping buttons and sending the creator a message.

Connecting to real data will come in a future PR.

# 🛠 How

Views are implemented as SwiftUI views, broken up into a few pieces, including individual flags, a project details view, a creator summary view, an address summary view, and buttons. Where possible, views used existing styling functions but did not use Prelude.

Snapshot testing is already complete and will be included in the card view PR.

# 👀 See

[Figma](https://www.figma.com/design/r0WKSECmMTCJTSKQskt2SQ/Backed-Projects-Dashboard?node-id=722-8194&t=rpm5RzgK15wT9Peu-0)

# ♿️ Accessibility 

- [x] Tap targets use minimum of 44x44 pts dimensions

Should support Dynamic Type and VoiceOver, but that hasn't been confirmed

# 🏎 Performance

Performance will be validated when these are integrated into the pledged project list.

# ✅ Acceptance criteria

Feature will be more thoroughly tested against real data in a future PR after integrating. For now, all tests should pass.

# ⏰ TODO

Future PRs will integrate into the app UI itself and more testing can happen at that time.

### CODE REVIEW

File: Kickstarter-iOS/Features/PledgedProjectsOverview/CardView/PPOAlertFlag.swift [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    }
+  }
+
+  struct Alert: Identifiable {
+    let type: AlertType
+    let icon: AlertIcon
+    let message: String
+
+    var id: String {
+      "\(self.type)-\(self.icon)-\(self.message)"
```
Comment (ankushkushwaha) [inline Kickstarter-iOS/Features/PledgedProjectsOverview/CardView/PPOAlertFlag.swift:61]: The alert id can be same for same message, is this what we are tying to do here?

File: Kickstarter-iOS/Features/PledgedProjectsOverview/CardView/PPOAlertFlag.swift [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+  var backgroundColor: Color {
+    switch self.alert.icon {
+    case .warning:
+      Color(uiColor: Constants.warningBackgroundColor)
+    case .alert:
+      Color(uiColor: Constants.alertBackgroundColor)
+    }
+  }
+
+  struct Alert: Identifiable {
```
Comment (ifosli) [inline Kickstarter-iOS/Features/PledgedProjectsOverview/CardView/PPOAlertFlag.swift:55]: We discussed this offline, but I do think this should live in a different file. I'm okay with this living here for now if it's moving once it's using real data, though!

**Review by ifosli — APPROVED** [review]
Couple minor things but overall looks good!

File: Kickstarter-iOS/Features/PledgedProjectsOverview/CardView/PPOAlertFlag.swift [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+    static let alertForegroundColor = UIColor.hex(0x73140D)
+    static let alertBackgroundColor = UIColor.hex(0xFEF2F1)
+
+    static let timeImage = ImageResource.iconLimitedTime
+    static let alertImage = ImageResource.iconNotice
+
+    static let imageSize: CGFloat = 18
+    static let spacerWidth: CGFloat = 4
+    static let cornerRadius: CGFloat = 6
```
[the remark this answers is not in the document]
Comment (stevestreza-ksr) [reply Kickstarter-iOS/Features/PledgedProjectsOverview/CardView/PPOAlertFlag.swift:105]: After discussing with Alison, we're changing this to 4, so I'm switching this to `Styles.cornerRadius`.

File: Kickstarter-iOS/Features/PledgedProjectsOverview/CardView/PPOAlertFlag.swift [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+      }
+    }
+  }
+
+  private enum Constants {
+    static let warningForegroundColor = UIColor.ksr_support_400
+    static let warningBackgroundColor = UIColor.ksr_celebrate_100
+
+    static let alertForegroundColor = UIColor.hex(0x73140D)
+    static let alertBackgroundColor = UIColor.hex(0xFEF2F1)
```
[the remark this answers is not in the document]
Comment (stevestreza-ksr) [reply Kickstarter-iOS/Features/PledgedProjectsOverview/CardView/PPOAlertFlag.swift]: After discussing with Alison these colors will be unique to here for right now but will probably graduate to being part of the design system eventually.

I've moved the constants for these files out into a separate `PPOCardStyles` and moved the constants for the other views into there.

**Comment by nativeksr** [comment]
### SwiftFormat found issues:

| File | Rules |
| ---- | ----- |
| Kickstarter-iOS/Features/PledgedProjectsOverview/CardView/PPOProjectCreator.swift:13:1 | warning: (redundantSelf) Insert/remove explicit self where applicable. |
| Kickstarter-iOS/Features/PledgedProjectsOverview/CardView/PPOProjectCreator.swift:14:1 | warning: (redundantSelf) Insert/remove explicit self where applicable. |
| Kickstarter-iOS/Features/PledgedProjectsOverview/CardView/PPOProjectCreator.swift:15:1 | warning: (redundantSelf) Insert/remove explicit self where applicable. |
| Kickstarter-iOS/Features/PledgedProjectsOverview/CardView/PPOProjectCreator.swift:15:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |
| Kickstarter-iOS/Features/PledgedProjectsOverview/CardView/PPOProjectCreator.swift:16:1 | warning: (redundantSelf) Insert/remove explicit self where applicable. |
| Kickstarter-iOS/Features/PledgedProjectsOverview/CardView/PPOProjectCreator.swift:16:1 | warning: (wrap) Wrap lines that exceed the specified maximum width. |
| Kickstarter-iOS/Features/PledgedProjectsOverview/CardView/PPOProjectCreator.swift:24:1 | warning: (redundantSelf) Insert/remove explicit self where applicable. |
| Kickstarter-iOS/Features/PledgedProjectsOverview/CardView/PPOProjectCreator.swift:25:1 | warning: (redundantSelf) Insert/remove explicit self where applicable. |
| Kickstarter-iOS/Features/PledgedProjectsOverview/CardView/PPOProjectCreator.swift:26:1 | warning: (redundantSelf) Insert/remove explicit self where applicable. |
| Kickstarter-iOS/Features/PledgedProjectsOverview/CardView/PPOProjectCreator.swift:27:1 | warning: (redundantSelf) Insert/remove explicit self where applicable. |
| Kickstarter-iOS/Features/PledgedProjectsOverview/CardView/PPOProjectCreator.swift:30:1 | warning: (redundantSelf) Insert/remove explicit self where applicable. |
| Kickstarter-iOS/Features/PledgedProjectsOverview/CardView/PPOProjectCreator.swift:35:1 | warning: (redundantSelf) Insert/remove explicit self where applicable. |
| Kickstarter-iOS/Features/PledgedProjectsOverview/CardView/PPOProjectCreator.swift:36:1 | warning: (redundantSelf) Insert/remove explicit self where applicable. |
| Kickstarter-iOS/Features/PledgedProjectsOverview/CardView/PPOProjectCreator.swift:37:1 | warning: (redundantSelf) Insert/remove explicit self where applicable. |


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Kickstarter-iOS/Features/PledgedProjectsOverview/CardView/PPOAlertFlag.swift (hunk 1 above)
Comment (stevestreza-ksr) [reply Kickstarter-iOS/Features/PledgedProjectsOverview/CardView/PPOAlertFlag.swift:61]: Our schema doesn't have an identifier but we shouldn't get duplicates from the API anyway.

**Pull request merged**